### ISSUE TITLE
ksh: Unexpected failure when using functions in arithmetic expressions

### ISSUE THREAD

**Issue by JasonGantner** [koalaman/shellcheck#2999]
#### For bugs
- Rule Id (if any, e.g. SC1000): N/A
- My shellcheck version (`shellcheck --version` or "online"): 0.10.0
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [X] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/ksh
h=22
m=34
INTERVAL=120
  ((count = floor(((23 - h) * 60 + (59 - m)) * (60.0 / INTERVAL))))
echo $count
```

#### Here's what shellcheck currently says:
```
  ^-- SC1073 (error): Couldn't parse this ((..)) command. Fix to allow more checks.
  ^-- SC1009 (info): The mentioned syntax error was in this ((..)) command.
                 ^-- SC1072 (error):  Fix any mentioned problems and try again.
```

#### Here's what I wanted or expected to see:

No error

##

ksh supports the following builtin arithmetic functions from the C math library:

```
abs acos acosh asin asinh atan atan2 atanh cbrt ceil copysign cos cosh erf erfc exp exp10 exp2 expm1 fabs fdim
finite float floor fma fmax fmin fmod fpclass fpclassify hypot ilogb  int  isfinite  isgreater isgreaterequal
isinf isinfinite isless islessequal islessgreater isnan isnormal issubnormal isunordered iszero j0 j1 jn ldexp
lgamma log log10 log1p log2 logb nearbyint nextafter nexttoward pow remainder rint round scalb scalbn  signbit
sin sinh sqrt tan tanh tgamma trunc y0 y1 yn
```

see `man 1 ksh` for details or `man <(curl -LJ https://github.com/ksh93/ksh/blob/dev/src/cmd/ksh93/sh.1)`

On a side note, zsh (which is unsupported, I know) has similar support given the module zsh/mathfunc is loaded
